File: 973. K Closest Points to Origin/solution.java
class Solution {
    class Compare implements Comparator<int[]> {
        public int compare(int[] a, int[] b) {
            return -(a[0] * a[0] + a[1] * a[1]) + (b[0] * b[0] + b[1] * b[1]);
        }
    } 
    
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<int[]> pq = new PriorityQueue<>(new Compare());
        for (int i = 0; i < points.length; i++) {
            pq.add(points[i]);
            if (pq.size() > k) {
                pq.poll();
            }
        }
        int[][] results = new int[k][2];
        int i = 0;
        while (!pq.isEmpty()) {
            int[] point = pq.poll();
            results[i++] = point;
        }
        return results;
    }
}

File: 973. K Closest Points to Origin/solution.py
import heapq
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        data = []
        for point in points:
            data.append((-(point[0] * point[0] + point[1] * point[1]), point))
        heap = []
        for d in data:
            heapq.heappush(heap, d)
            if len(heap) > k:
                heapq.heappop(heap)
        ret = []
        for h in heap:
            ret.append(h[1])
        return ret
